core/src/test/java/org/keycloak/jose/JsonWebTokenTest.java

```diff
@@ -59,4 +61,31 @@ public void testArray() throws IOException {
         assertTrue(JsonSerialization.writeValueAsPrettyString(jsonWebToken).contains("\"aud\" : [ \"test\", \"test2\" ]"));
     }
 
+    @Test
+    public void isActiveReturnFalseWhenBeforeTimeInFuture() throws IOException {
+        int currentTime = Time.currentTime();
+        int futureTime = currentTime + 1;
+        JsonWebToken jsonWebToken = new JsonWebToken();
+        jsonWebToken.notBefore(futureTime);
+        assertFalse(jsonWebToken.isActive());
+    }
+
+    @Test
+    public void isActiveReturnTrueWhenBeforeTimeInPast() throws IOException {
+        int currentTime = Time.currentTime();
+        int pastTime = currentTime - 1;
+        JsonWebToken jsonWebToken = new JsonWebToken();
+        jsonWebToken.notBefore(pastTime);
+        assertTrue(jsonWebToken.isActive());
+    }
+
+    @Test
+    public void isActiveShouldAllowTimeSkewWhenBeforeTimeInPast() throws IOException {
+        int currentTime = Time.currentTime();
+        int pastTime = currentTime - 50;
```

Comment:
Isn't this testing the opposite case that you actually wanted? It seems the difference will be 101, not 1? And if I understand correctly, the point of this would be to test if the difference is only 1?

Reply:
Perhaps i should change that the past time is just 1 second it past, but here i chose 51 seconds as tolerable amount of time difference.
The idea of test was setting the not before time 50 seconds in past and assert that with tolerable clock skew = 51 the token is considered active.

Reply:
What I meant is, that it's the opposite case that should be tested? If you want to test that this works "jsonWebToken.isActive(51)", you may need to set "notBefore" flag of jsonWebToken 50 seconds to the future, not to the past?

Reply:
You were absolutely right. Setting not before time to past does not make sense as it is valid anyway without clock skew tolerating.
Changed the unit tests and added the scenario when clock is off for more than tolerable clock skew.